### DISCUSSION TITLE
run (ok), move & goto (!!)

### DISCUSSION THREAD

**Discussion by 7sorkr** [ponoor/step-series-support#19] [Q&A] [answered]
STEP400 is in use.

in OSC
The run signal is fine.

1. Move and goto will boot if you give a different signal while moving. (4 red blinks)

I have one more question for you.

2. like gountil signal
We ask if it is possible to stop if it is located on the home sensor while moving by run, move, and goto signals.

thank you

**Comment by kanta** [comment, marked as answer]
Hi @7sorkr , 

I need some more information from you to know what happened. Could you check the firmware version of your STEP400 board? Here is an instruction for checking the firmware version.
https://ponoor.com/en/docs/step-series/technical-information/firmware-update/

Also, could you tell me what kind of commands you sent during `/move` or `/goTo`?

To stop the motor when the home sensor activated, you can use [`/setHomeSwMode`](https://ponoor.com/en/docs/step-series/osc-command-reference/home-and-limit-sensers/#sethomeswmode_intmotorid_boolsw_mode) command. For example, you can send
```
/setHomeSwMode 1 0
```
then the home sensor 1 will always cause hard stop. I hope this works!

**Reply by 7sorkr** [reply]
============== Current Status ==============
-------------- Firmware --------------
Firmware name : STEP400
Firmware version : 1.0.0
Compile date : Mar 12 2021, 16:33:15
Applicable config version : 1.0
Loaded config version : 1.2 [CONFIG_VERSION_NEW]
-------------- DIP Switch --------------
BIN : 0000 0001
DEC : 1
-------------- Ethernet --------------
Ethernet hardware status: 3 -EthernetW5500
Ethernet link status: 1 -LinkON
isDestIpSet : Yes
-------------- microSD --------------
SD library initialize succeeded : Yes
SD config file open succeeded : Yes
SD config JSON parse succeeded : Yes
-------------- PowerSTEP01 --------------
PowerSTEP01 SPI connection established : Yes
PowerSTEO01 ID#1
	STATUS: 0xE603
	High impedance state : Yes
	BUSY : No
	Motor direction : Reverse
	Motor status : Stopped
	UVLO (Undervoltage lock out) : No
	Thermal status : Normal
	OCD (Overcurent detection) : No
	Stalled : No
	SW_F: 0 -HOME senser input open.
	ADC_OUT: 31 -LIMIT senser input open.
PowerSTEO01 ID#2
	STATUS: 0xE603
	High impedance state : Yes
	BUSY : No
	Motor direction : Reverse
	Motor status : Stopped
	UVLO (Undervoltage lock out) : No
	Thermal status : Normal
	OCD (Overcurent detection) : No
	Stalled : No
	SW_F: 0 -HOME senser input open.
	ADC_OUT: 31 -LIMIT senser input open.
PowerSTEO01 ID#3
	STATUS: 0xE603
	High impedance state : Yes
	BUSY : No
	Motor direction : Reverse
	Motor status : Stopped
	UVLO (Undervoltage lock out) : No
	Thermal status : Normal
	OCD (Overcurent detection) : No
	Stalled : No
	SW_F: 0 -HOME senser input open.
	ADC_OUT: 31 -LIMIT senser input open.
PowerSTEO01 ID#4
	STATUS: 0xE603
	High impedance state : Yes
	BUSY : No
	Motor direction : Reverse
	Motor status : Stopped
	UVLO (Undervoltage lock out) : No
	Thermal status : Normal
	OCD (Overcurent detection) : No
	Stalled : No
	SW_F: 0 -HOME senser input open.
	ADC_OUT: 31 -LIMIT senser input open.
-------------- Modes --------------
Servo Mode :  No, No, No, No
Current Mode :  No, No, No, No
Electromagnetic Brake Enable :  No, No, No, No
Brake status :
#1 : BRAKE_ENGAGED
#2 : BRAKE_ENGAGED
#3 : BRAKE_ENGAGED
#4 : BRAKE_ENGAGED
Homing status : 0, 0, 0, 0

**Reply by 7sorkr** [reply]
Firmware 1.2.1 update has been completed.

Arduino SAMD Boards (32-bits ARM Cortex-M0+)
I installed it with Ver 1.8.10 and it worked fine.

**Reply by kanta** [reply]
Great! Have all the problems been resolved?

**Reply by 7sorkr** [reply]
Each OSC in the document is being tested.

Thank you, kanta.
